### PR TITLE
Zmq isolation

### PR DESCRIPTION
**Pull request by phlptp**
### Summary

If merged this pull request will merge all the zmq related files together and clean up the CMake code to use targets instead of putting the library name in a variable.  As a first step in shifting the ZMQ library stuff out of the main CMake file.  

### Proposed changes

- move zmq related items out of the common library.  
- update the header locations 
- change the cmake related items for ZMQ.

### CODE REVIEW

**Comment by phlptp** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones added
============
- src/helics/apps/zmqBrokerServer.cpp  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/GMLC-TDC/HELICS/pullRequest?prid=5423186&bid=17784270)

**Comment by phlptp** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones added
============
- src/helics/apps/zmqBrokerServer.cpp  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/GMLC-TDC/HELICS/pullRequest?prid=5423186&bid=17784270)

**Comment by phlptp** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones added
============
- src/helics/apps/zmqBrokerServer.cpp  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/GMLC-TDC/HELICS/pullRequest?prid=5423186&bid=17784270)

**Comment by phlptp** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones added
============
- src/helics/apps/zmqBrokerServer.cpp  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/GMLC-TDC/HELICS/pullRequest?prid=5423186&bid=17784270)

File: CMakeLists.txt [hunk 1]
```diff
@@ -514,15 +514,15 @@ if(ENABLE_ZMQ_CORE)
     if(NOT ZeroMQ_FOUND)
         message(FATAL_ERROR "ZeroMQ not found, needed to enable the ZMQ Core")
     endif()
-
+	add_library(zmq INTERFACE)
     if(HELICS_USE_ZMQ_STATIC_LIBRARY)
-        set(ZeroMQ_DEPENDENCY libzmq-static)
-        target_compile_definitions(helics_base INTERFACE -DZMQ_STATIC)
+		target_link_libraries(zmq INTERFACE libzmq-static)
+        target_compile_definitions(zmq INTERFACE -DZMQ_STATIC)
```
Comment (nightlark) [inline CMakeLists.txt:520]:
```suggestion
        target_link_libraries(zmq INTERFACE libzmq-static)
        target_compile_definitions(zmq INTERFACE -DZMQ_STATIC)
```

File: CMakeLists.txt [hunk 2]
```diff
@@ -514,15 +514,15 @@ if(ENABLE_ZMQ_CORE)
...
     else()
-        set(ZeroMQ_DEPENDENCY libzmq)
+		target_link_libraries(zmq INTERFACE libzmq)
```
Comment (nightlark) [inline CMakeLists.txt:522]:
```suggestion
        target_link_libraries(zmq INTERFACE libzmq)
```

File: config/cmake/addBoost.cmake [hunk 3]
```diff
@@ -15,6 +15,7 @@ if(WIN32 AND NOT UNIX_LIKE)
 
     set(
         boost_versions
+		boost_1_73_0
         boost_1_72_0
```
Comment (nightlark) [inline config/cmake/addBoost.cmake:19]:
```suggestion
        boost_1_73_0
        boost_1_72_0
```

File: ThirdParty/cppzmq/zmq.hpp [hunk 4]
```diff
@@ -297,6 +297,11 @@ inline int poll(std::vector<zmq_pollitem_t> &items, long timeout_ = -1)
 }
 #endif
 
+inline bool has_message(const zmq_pollitem_t &item)
+{
+    return ((static_cast<uint32_t>(item.revents) & ZMQ_POLLIN)!=0U);
+}
+
```
Comment (nightlark) [inline ThirdParty/cppzmq/zmq.hpp:304]: Is this something in the upstream version of cppzmq?

Comment (phlptp) [reply ThirdParty/cppzmq/zmq.hpp:304]: It cleared up a clang-tidy warning and simplified some code.  So it isn't in the upstream, but there are a few other functions we have like that as well.   I wanted to have it be inline and it needed to have access to zmq.h which isn't true for a lot of the other headers so this seemed like the most logical place at the moment.

**Comment by phlptp** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones added
============
- src/helics/apps/zmqBrokerServer.cpp  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/GMLC-TDC/HELICS/pullRequest?prid=5423186&bid=17784270)

**Pull request merged**